### ISSUE TITLE
[DISCUSS] Dropping the task audit log table from metastore

### ISSUE THREAD

**Issue by jihoonson** [apache/druid#5859]
Druid currently has a table to store task audit logs in metastore. Every taskAction is stored in this table if they need to be audited (https://github.com/druid-io/druid/blob/master/indexing-service/src/main/java/io/druid/indexing/common/actions/TaskAction.java#L47). Basically, all taskActions modifying metastore like `SegmentInsertAction` currently need to be audited. If a taskAction needs to be audited, it is serialized into a byte array and stored as a blob in metastore.

In Druid, there is a single use case for the task audit log table: an overlord API to get a list of segments generated by a task ([http://<OVERLORD_IP>:<port>/druid/indexer/v1/task/{taskId}/segments](http://druid.io/docs/latest/design/indexing-service.html#submitting-tasks-and-querying-task-status)). This API needs to find all `SegmentInsertAction`s and `SegmentTransactionalInsertAction`s performed by the given task. AFAIK, this API isn't used widely. Outside of Druid, AFAIK, there is no good use case for task audit logs table.

The problem is that this table can cause an issue when inserting a large taskAction. For example, a `SegmentInsertAction` can be large if there are many segments to be inserted in that action. This can cause some issues. For example, in MySQL, it can cause the insertion error if the data size is larger than [max_allowed_packet](https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_max_allowed_packet).

I tried to mitigate this in https://github.com/druid-io/druid/pull/5751, but looks not a good idea. Another suggestion in that PR was to store audit logs in deep storage, but it requires quite a lot of changes and introduces new design issues. I don't think we need to make a large effort to fix such an issue that has no good use case.

So, I'm suggesting dropping the task audit log table. Here is my suggestion.

1) Disabling task audit logging by default in 0.13. We need to add a configuration for optionally enabling task audit logging. The overlord API should also be deprecated.
2) Completely removing the task audit log table and the overlord API in 0.14.

Any thoughts?

**Comment by zhangxinyu1** [comment]
The only usage of the audit table in our company is recording the information about who and when change the rules of datasource.

**Comment by jihoonson** [comment]
@zhangxinyu1 thanks for the comment. The audit logs for changing load/drop rules are stored in another table, named `druid_audit` by default. The table issued here is `druid_tasklogs` table which stores audit logs for changing segments.

**Comment by himanshug** [comment]
I don't remember needing to use that table ever since I have used Druid. So, from my side I think it is ok to remove it.

**Comment by nishantmonu51** [comment]
+1 on dropping task action audit table. Never seen anyone using that.

**Comment by jihoonson** [comment]
#6368 has been merged to master. Will raise a follow-up pr to drop task audit log table for 0.14 later.

**Comment by KenjiTakahashi** [comment]
I see I'm kind of late to the party here ;-). We are using the `/segments` endpoint internally for validation purposes. I.e. we have long running Tasks with EventReceiverFirehose and the service that pushes data, then checks if appropriate segments/shards got created by the Task.

**Comment by jihoonson** [comment]
Would you please give us more details about how you use ‘/segments’ for validation? I’m not sure what you mean by the validation. How do you know what segments are expected to be generated?

**Comment by KenjiTakahashi** [comment]
Sorry about such late response! I'm not working on this anymore, and it kinda slipped my attention...
Anyway, AFAIR, it works basically by knowing how much data will be sent and how it will be split into shards. After sending the data, it surely uses `/segments` to check if correct number of shards got created, and perhaps some other details, as well.
It works with multiple Tasks in parallel, independently, so it all has to be done at Task level.
/cc @neutrinus?

**Comment by jihoonson** [comment]
@KenjiTakahashi thanks but I still don't understand how it works.. And also I'm wondering how useful your validation is. Is it very important to create the expected number of segments in your use case?

**Comment by jihoonson** [comment]
This issue still relevant.
